controllers/domain/getDomain.js

```diff
@@ -68,7 +68,11 @@ module.exports = async (req, res) => {
             }
           }
 
-          if (item?.domain?.credderScore >= CREDDER_MIN_SCORE || !CREDDER_CHECK_ENABLED) {
+          const isNewsExist = data.find((news) => news.content.news_url === item.content.news_url);
+          if (
+            (item?.domain?.credderScore >= CREDDER_MIN_SCORE || !CREDDER_CHECK_ENABLED) &&
+            !isNewsExist
+          ) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for checking if a news item already exists in the data array and preventing duplicates is correct. However, using `Array.prototype.find()` in a loop can lead to performance issues when dealing with large datasets because it has a time complexity of O(n). A more efficient approach would be to use a Set or a Map to store the URLs of the news items, which allows constant-time lookup.

Here's how you could modify your code:

```diff
+    const newsUrls = new Set(data.map(news => news.content.news_url));
     const isNewsExist = newsUrls.has(item.content.news_url);
     if (
       (item?.domain?.credderScore >= CREDDER_MIN_SCORE || !CREDDER_CHECK_ENABLED) &&
       !isNewsExist
     ) {
+      newsUrls.add(item.content.news_url);
       data.push(item);
     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->